src/OrleansRuntime/Catalog/GrainLifecycle.cs

```diff
@@ -0,0 +1,10 @@
+﻿
+namespace Orleans.Runtime
+{
+    internal class GrainLifecycle : LifecycleObservable<GrainLifecyleStage>, IGrainLifeCycle
```

Comment:
Is LifecycleObservable a bit heavyweight for this purpose? It's built for multi-threaded access and the implementation is quite allocation happy. If it was just being used for silo startup, then that's fine, but for every single activation it seems a bit heavy